lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -1520,6 +1562,26 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
               "Scope of the debug variable should have the same parent function"
               " as that of instruction.");
 
+    llvm::StringRef UniqueName = VarNames.insert(varInfo->Name).first->getKey();
```

Comment:
Can you add a comment here explaining in broad strokes what this check is doing?